pkg/kubelet/volume/empty_dir/empty_dir.go

```diff
@@ -99,8 +158,20 @@ func (ed *emptyDir) SetUp() error {
 	if ed.legacyMode {
 		return fmt.Errorf("legacy mode: can not create new instances")
 	}
-	path := ed.GetPath()
-	return os.MkdirAll(path, 0750)
+	if ed.tmpfs {
+		return ed.setupTmpfs()
+	}
+	return os.MkdirAll(ed.GetPath(), 0750)
+}
+
+func (ed *emptyDir) setupTmpfs() error {
+	if ed.mounter == nil {
+		return fmt.Errorf("tmpfs requested, but mounter is nil")
+	}
+	if err := os.MkdirAll(ed.GetPath(), 0750); err != nil {
+		return err
+	}
+	return ed.mounter.Mount("tmpfs", ed.GetPath(), "tmpfs", 0, "")
```

Comment:
We can potentially also add MkdirAll() to Mount() since all current users of it do the same logic.